app/serializers/concerns/topic_tags_mixin.rb

```diff
@@ -3,23 +3,37 @@
 module TopicTagsMixin
   def self.included(klass)
     klass.attributes :tags
+    klass.attributes :tags_descriptions
   end
 
   def include_tags?
     scope.can_see_tags?(topic)
   end
 
   def tags
-    # Calling method `pluck` or `order` along with `includes` causing N+1 queries
-    tags = (SiteSetting.tags_sort_alphabetically ? topic.tags.sort_by(&:name) : topic.tags.sort_by(&:topic_count).reverse).map(&:name)
-    if scope.is_staff?
-      tags
-    else
-      tags - scope.hidden_tag_names
-    end
+    all_tags.map(&:name)
+  end
+
+  def tags_descriptions
+    all_tags.reduce({}) do |acc, tag|
+      acc[tag.name] = tag.description
+      acc
+    end.compact
```

Comment:
Just and FYI/TIL, since `with_object` changed my life – no more hanging variables just to return the correct thing!

```suggestion
    all_tags.each.with_object({}) { |tag, acc| acc[tag.name] = tag.description }.compact
```

Reply:
Ohh nice!